browser/resources/settings/default_brave_shields_page/components/brave_adblock_scriptlet_editor.ts

```diff
@@ -127,17 +127,19 @@ class AdblockScriptletEditor extends AdblockScriptletEditorBase {
   }
 
   validateName_() {
-    this.scriptlet.name = this.scriptlet.name.toLowerCase()
     if (!/^[a-zA-Z0-9-_.]+$/.test(this.scriptlet.name)) {
       this.updateError(ErrorCode.kInvalidName)
     } else {
       this.updateError(ErrorCode.kOK)
     }
   }
 
+  startsWithCaseInsensitive(str: string, prefix: string): boolean {
+    return str.slice(0, prefix.length).toLowerCase() == prefix.toLowerCase()
+  }
+
   updateScriptletBeforeSave_() {
-    this.scriptlet.name = this.scriptlet.name.toLowerCase()
-    if (!this.scriptlet.name.startsWith('user-')) {
+    if (!this.startsWithCaseInsensitive(this.scriptlet.name, 'user-')) {
```

Comment:
Why do we need this?

Reply:
To be sure that user's scriptlets starts with `user-` prefix